agenthub/codeact_agent/codeact_agent.py

```diff
@@ -205,21 +205,46 @@ def step(self, state: State) -> Action:
         # prepare what we want to send to the LLM
         messages = self._get_messages(state)
 
-        response = self.llm.completion(
-            messages=[message.model_dump() for message in messages],
-            stop=[
+        params = {
+            'messages': [message.model_dump() for message in messages],
+            'stop': [
                 '</execute_ipython>',
                 '</execute_bash>',
                 '</execute_browse>',
             ],
-            temperature=0.0,
-        )
+            'temperature': 0.0,
+        }
+
+        if self.llm.supports__prompt_caching():
+            params['extra_headers'] = {
+                'anthropic-version': '2023-06-01',
```

Comment:
Just wondering, is this header necessary? In their docs, Anthropic says that the beta header is needed (the other one). In their examples they add the api version too, but that might be because it's required in general, in which case I would assume litellm is doing this in all requests, not just prompt caching. I could be wrong though. 🤔

Reply:
You are correct I think, it worked without it, also the vesion is super old. Thanks for noticing. The version was present in the examples, weird.